packages/api/src/controllers/helpers.ts

```diff
@@ -222,8 +223,9 @@ export async function getObjectStoreS3Config(
     region,
     bucket,
     signingRegion: region,
-    endpoint: `${protocol}//${url.host}`,
+    endpoint: `${protocol}//${url.host}/${bucket}`,
     forcePathStyle: true,
+    s3BucketEndpoint: true,
```

Comment:
Where does this field come from and what's the difference from `forcePathStyle`? If I try this locally I get that the field does not exist in this type, maybe that's why CI is failing as well.

Reply:
I found the documentation here <https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/S3.html> and just verified that flipping it off reintroduces the problem. Do you need to add it to the type just above that function in your local setup?

CI's failing because presigned URLs are now getting signed with `/bucket-name/bucket-name`

Reply:
It doesn't need to be added to the `OSS3Config` type cause we already "contains" `S3ClientConfig`. Maybe the version needs to be updated?

> CI's failing because presigned URLs are now getting signed with /bucket-name/bucket-name

I think this would happen with only your change to the endpoint above. I'm suspecting this new flag is being a noop

Reply:
Just tested it again, `s3BucketEndpoint: false` definitely reintroduces the problem. Maybe the types are out-of-date?

Reply:
@iameli-streams maybe tus is using a different S3 library than ourselves 👀